infrastructure/modules/postgres_server/main.tf

```diff
@@ -0,0 +1,73 @@
+data "azurerm_client_config" "current" {}
+
+data "azurerm_resource_group" "postgres_server" {
+  name = var.resource_group_name
+}
+
+# https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#security
+data "azurerm_role_definition" "key_vault_crypto_officer" {
+  role_definition_id = "14b46e9e-c2b7-41b4-b07b-48a6ebf60603"
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/user_assigned_identity
+resource "azurerm_user_assigned_identity" "postgres_server" {
+  name                = "mipsqlsrv${var.metadata.suffix}"
+  resource_group_name = data.azurerm_resource_group.postgres_server.name
+  location            = data.azurerm_resource_group.postgres_server.location
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/role_assignment
+resource "azurerm_role_assignment" "key_vault_crypto_officer" {
```

Comment:
Du har `key_vault_crypto_officer` baade som `data` og `resource` - dette er vel et tegn paa at data-en ikke trengs?

Reply:
data er ikke nødvendig, men foretrekker å slå opp rollen med guid før jeg bruker navnet på rollen direkte (dersom atter hvis om atter navnet skulle endre, noe det ikke skal gjøre!). Har sett at man bruker dette en del steder ved bruk av azurerm_role_assignment.

Reply:
Kan du ikke faa ut guiden av resursen? Mulig jeg blander arm og terraform naa bare saa det er sagt...

Reply:
mapper role_definition_id istedenfor å bruke role_definition_name. Husker at det har vært et issues med å bruke definition id for denne ressursen tidligere, husker ikke nødvendigvis hva som var greia, er noen år siden :p